dora/core/server/master/src/main/java/alluxio/master/job/DoraLoadJob.java

```diff
@@ -147,74 +145,27 @@ public DoraLoadJob(
     mBandwidth = bandwidth;
     mUsePartialListing = usePartialListing;
     mVerificationEnabled = verificationEnabled;
-    FileSystem fs = FileSystem.Factory.create(Scheduler.getInstance().getFileSystemContext());
-    ListStatusPOptions listOptions = ListStatusPOptions
-        .newBuilder()
-        .setRecursive(true).build();
-    mFs = fs;
-    mFileIterator = Optional.of(new FileListFetcher(fs, mPath, listOptions, false));
-  }
+    String ufsRoot = Configuration.getString(PropertyKey.DORA_CLIENT_UFS_ROOT);
+    AlluxioURI ufsRootUri = new AlluxioURI(ufsRoot);
+    mUfs = UnderFileSystem.Factory.create(
+        ufsRoot,
+        UnderFileSystemConfiguration.defaults(Configuration.global()));
 
-  private static class FileListFetcher implements Iterator<URIStatus> {
-    public static final int PREFETCH_SIZE = 1000;
-    private final FileSystem mFs;
-    private String mPath;
-    ListStatusPOptions mListOption;
-    public LinkedBlockingQueue<URIStatus> mFiles = new LinkedBlockingQueue<>();
-    private boolean mUsePartialListing = false;
-    private AtomicBoolean mIsDone = new AtomicBoolean(false);
-
-    public FileListFetcher(FileSystem fs, String path, ListStatusPOptions listOption,
-                           boolean usePartialListing) {
-      mPath = path;
-      mListOption = listOption;
-      mFs = fs;
-      mUsePartialListing = false; // default to false now
-    }
-
-    private int advance() {
-      if (mIsDone.get()) {
-        return 0;
-      }
-      try {
-        // TODO(lucy) paginate list a PREFETCH_SIZE here.
-        List<URIStatus> uriStatuses = mFs.listStatus(new AlluxioURI(mPath), mListOption);
-        if (uriStatuses == null) {
-          mIsDone.compareAndSet(false, true);
-          return 0;
-        }
-        uriStatuses.forEach(uriStatus -> mFiles.offer(uriStatus));
-        if (!mUsePartialListing) {
-          mIsDone.compareAndSet(false, true);
-        }
-        return uriStatuses.size();
-      } catch (IOException | AlluxioException e) {
-        throw AlluxioRuntimeException.from(e);
-      }
-    }
+    System.out.println("Active workers: " + Scheduler.getInstance().getActiveWorkers().size());
 
-    @Override
-    public boolean hasNext() {
-      while (!mIsDone.get() && mFiles.size() < PREFETCH_SIZE * 0.2) {
-        if (advance() <= 0) {
-          break;
-        }
-      }
-      return mFiles.peek() != null;
-    }
-
-    @Override
-    public URIStatus next() {
-      while (!mIsDone.get() && mFiles.size() < PREFETCH_SIZE * 0.2) {
-        if (advance() <= 0) {
-          break;
-        }
-      }
-      URIStatus uriStatus = mFiles.poll();
-      if (uriStatus == null) {
-        throw new NoSuchElementException();
+    try {
+      mUfsStatusIterator = mUfs.listStatusIterable(
+          path, ListOptions.defaults().setRecursive(true), null, 0);
```

Comment:
plans to support partial listing?

Reply:
is this path supposed to be alluxio path or ufs path?

Reply:
this is supposed to be ufs path...

partial listing is supported by the interface but is not used in the scheduler